daffodil-core/src/main/scala/org/apache/daffodil/dsom/Term.scala

```diff
@@ -297,31 +317,31 @@ trait Term
     res
   }
 
+  /**
+   * One-based position in the nearest enclosing sequence.
+   * Follows backpointers from group defs to group refs until it
+   * finds a sequence.
+   */
   final lazy val positionInNearestEnclosingSequence: Int = {
```

Comment:
Classic example of a method that creates a pointless need for the backpointers to parent that make structure sharing of the DSOM objects impossible. This should be a value passed down to a SequenceChild constructor, and the algorithms that use this should be on SequenceTermBase or child classes thereof.